lib/src/component/http_body.rs

```diff
@@ -168,12 +168,17 @@ impl http_body::Host for Session {
             b'\0',
             usize::try_from(max_len).unwrap(),
             cursor,
-        );
-        if buf.is_empty() && next.is_none() {
-            return Ok(None);
-        }
+        )
+        .map_err(|needed| types::Error::BufferLen(u64::try_from(needed).unwrap_or(0)))?;
 
-        Ok(Some((buf, next)))
+        // At this point we know that the buffer being empty will also mean that there are no
+        // remaining entries to read.
+        if buf.is_empty() {
+            debug_assert!(next.is_none());
```

Comment:
Do we have a test that runs over this case? Asserts always make me nervous.

Reply:
This was more just a guard against the future, for changes to `write_values`. I think it's impossible for `buf.is_empty()` to be true if `next.is_none()` is false, as that would be caught by early return in the loop in `write_values`. Additionally, writing a single empty entry into the result would cause at least the terminator byte `\0` to be written and the buffer to be non-empty.